docs_src/design/adr/security/0009-Secure-Bootstrapping.md

```diff
@@ -0,0 +1,295 @@
+[separate ADR](https://github.com/edgexfoundry/edgex-docs/issues/296).)
+
+Activities that are best done in the initialization phase include the following:
+
+* Bootstrapping of crytographic secrets needed by the application.
+* Bootstrapping of database users and passwords.
+* Installation of database schema needed for application logic to function.
+* Initialization of authorization frameworks such as configuring RBAC or ACLs.
+* Other one-time initialization activities.
+
+Workarounds when an installation phase is not present include:
+
+* Perform initialization tasks manually,
+  and manually seed secrets into static configuration files.
+* Ship with known hard-coded secrets in static configuration files.
+* Start in an insecure configuration and remain that way.
+* Provision some secrets at runtime.
+
+EdgeX does not have a manual installation flow,
+and uses a combination of the last three approaches.
+
+The objective of this ADR is to define a framework for Docker-based initialization logic in EdgeX.
+This will enable the removal of certain hard-coded secrets in EdgeX
+and enable certain components (such as Consul) to be started in a secure configuration.
+These improvement are necessary pre-requisites to implementing microservice communication security.
+
+### History
+
+In previous releases, container startup sequencing has been
+primarily been driven by Consul service health checks
+backed healthcheck endpoints of particular services
+or by sentinel files placed in the file system
+when certain intialization milestones are reached.
+
+The implementation has been plagued by several issues:
+
+* Sentinel files are not cleaned up if the framework fails or is shut down.
+  Invalid state left over from previous instantiations of the framework
+  causes difficult-to-resolve race conditions.
+  (Implementation of this ADR will try to remove as many as possible,
+  focusing on those that are used to gate startup.
+  Some use of sentinel files may still be required to indicate
+  completion of initialization steps so that they are not re-done
+  if there is no API-based mechanism to determine if such
+  initialization has been completed.)
+
+* Consul healh checks are reported in a difficult-to-parse
+  JSON structure, which has lead to the creation of specialized
+  tools that are insensitive to libc implementations
+  used by different container images.
+
+* Consul is being used not only for service health, but for
+  [service location and configuration as well](https://github.com/edgexfoundry/edgex-docs/issues/279).
+  The requirement to synchronize framework startup for
+  the purpose of securely initializing Consul means
+  that a non-Consul mechanism must be used
+  to stage-gate EdgeX initialization.
+
+This last point is the primary motivator of this ADR.
+
+## Decision
+
+### Stage-gate mechanism
+
+The stage-gate mechanism must work in the following environments:
+
+* docker-compose in Linux on a single node
```

Comment:
I know his is nit-picky, but node is a k8s term. Maybe clarify and say node/system?